File: dpo_train/generation_output/cpp-java/HumanEval_13_greatest_common_divisor.java
### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
long greatest_common_divisor(long a, long b) {
    while (b) {
        long temp = a % b; a = b; b = temp;
    }
    return a;
}

### Java version

import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static long greatestCommonDivisor(long a, long b) {
        long result = 1;
        while (b) {
            long temp = a % b;
            a = b; b = temp;
            result = temp;
        }
        return result;
    }
}

class Solution {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        long a = in.nextLong();
        long b = in.nextLong();
        long result = Problem.greatestCommonDivisor(a, b);
        System.out.println(result);
    }
}

File: test/before_train/trainset_and_testset/py-cpp/HumanEval_13_greatest_common_divisor.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
long greatest_common_divisor(long a, long b) {
    long c = 1;
    while (a) {
        c = c * a % b;
        a = a / b;
    }
    return c;
}
int main() {
    auto candidate = greatest_common_divisor;
    assert(candidate((3), (7)) == (1));
    assert(candidate((10), (15)) == (5));
    assert(candidate((49), (14)) == (7));
    assert(candidate((144), (60)) == (12));
}
